src/deprecated_library/castleengine.h

```diff
@@ -356,8 +356,30 @@ extern void CGE_SetWalkNavigationMouseDragMode(int /*ECgeMouseDragMode*/ eMode);
 extern void CGE_SetVariableInt(int /*ECgeVariable*/ eVar, int nValue);
 extern int CGE_GetVariableInt(int /*ECgeVariable*/ eVar);
 
-extern void CGE_SetNodeFieldValue(const char *szNodeName, const char *szFieldName,
-                                  float fVal1, float fVal2, float fVal3, float fVal4);
+extern void CGE_SetNodeFieldValue_SFFloat(const char *szNodeName, const char *szFieldName, float value);
+extern void CGE_SetNodeFieldValue_SFDouble(const char *szNodeName, const char *szFieldName, double value);
+extern void CGE_SetNodeFieldValue_SFInt32(const char *szNodeName, const char *szFieldName, int value);
+extern void CGE_SetNodeFieldValue_SFBool(const char *szNodeName, const char *szFieldName, int value);
```

Comment:
Can we use `bool` in C, looks more obvious than `int`? 

I understand from https://www.geeksforgeeks.org/bool-in-c/ , https://github.com/gcc-mirror/gcc/blob/master/gcc/ginclude/stdbool.h that `#include <stdbool.h>` defines it. 

I know it's still just `int` underneath, but calling it `bool` makes the purpose obvious for anyone reading the C header.

I see the `CTypes` in Pascal ( https://www.freepascal.org/docs-html/current/rtl/ctypes/index-3.html ) also defines `CBool` to match C `bool`.